### PR TITLE
Sfdc crm entities

### PR DESCRIPTION
**Pull request by cdoroftei**

### CODE REVIEW

File: Source/Actions/Microsoft.Deployment.Actions.MsCrm/CrmCreateProfile.cs [hunk 1]
```diff
@@ -53,11 +54,19 @@ public class CrmCreateProfile : BaseAction
...
             string name = request.DataStore.GetValue("ProfileName") ?? "bpst-mscrm-profile";
             string kV = request.DataStore.GetValue("KeyVault");
-            string[] entities = request.DataStore.GetValue("Entities").Split(new[] {',', ' ', '\t'}, StringSplitOptions.RemoveEmptyEntries);
+            var entities = request.DataStore.GetValue("Entities").Split(new[] {',', ' ', '\t'}, StringSplitOptions.RemoveEmptyEntries).ToList();
+
+            var additionalObjects = request.DataStore.GetValue("AdditionalObjects");
+
+            if (!string.IsNullOrEmpty(additionalObjects))
+            {
+                string[] add = additionalObjects.Split(',');
```
Comment (raymondb1) [inline Source/Actions/Microsoft.Deployment.Actions.MsCrm/CrmCreateProfile.cs]: split by space, tab, too and remove empty entries

File: Source/Actions/Microsoft.Deployment.Actions.MsCrm/CrmGetEntities.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
+﻿using System;
```
Comment (raymondb1) [inline Source/Actions/Microsoft.Deployment.Actions.MsCrm/CrmGetEntities.cs]: Is this CrmGetEntities.cs really necessary? Can't you just rely on validate entities?

File: Source/Actions/Microsoft.Deployment.Actions.MsCrm/CrmValidateEntities.cs [hunk 3]
```diff
@@ -30,6 +30,14 @@ class CrmValidateEntities : BaseAction
             string organizationUrl = request.DataStore.GetValue("OrganizationUrl");
             string[] entities = request.DataStore.GetValue("Entities").Split(new[] { ',', ' ', '\t' }, StringSplitOptions.RemoveEmptyEntries);
 
+            var additionalObjects = request.DataStore.GetValue("AdditionalObjects");
+
+            if (!string.IsNullOrEmpty(additionalObjects))
+            {
+                string[] add = additionalObjects.Split(',');
```
Comment (raymondb1) [inline Source/Actions/Microsoft.Deployment.Actions.MsCrm/CrmValidateEntities.cs]: same comment as above about splitting

File: Source/Actions/Microsoft.Deployment.Actions.MsCrm/CrmValidateProfile.cs [hunk 4]
```diff
@@ -12,6 +12,7 @@
     using Microsoft.Deployment.Common.ActionModel;
     using Microsoft.Deployment.Common.Actions;
     using Microsoft.Deployment.Common.Helpers;
+    using System.Linq;
```
Comment (raymondb1) [inline Source/Actions/Microsoft.Deployment.Actions.MsCrm/CrmValidateProfile.cs]: avoid Linq

File: Source/Actions/Microsoft.Deployment.Actions.MsCrm/CrmValidateProfile.cs [hunk 5]
```diff
@@ -54,6 +55,15 @@ public class CrmValidateProfile : BaseAction
             string kV = request.DataStore.GetValue("KeyVault");
             string[] entities = request.DataStore.GetValue("Entities").Split(new[] {',', ' ', '\t'}, StringSplitOptions.RemoveEmptyEntries);
 
+            var additionalObjects = request.DataStore.GetValue("AdditionalObjects");
+
+            if (!string.IsNullOrEmpty(additionalObjects))
+            {
+                string[] add = additionalObjects.Split(',');
```
Comment (raymondb1) [inline Source/Actions/Microsoft.Deployment.Actions.MsCrm/CrmValidateProfile.cs]: same comment as above about splitting. Worth making this a helper method

File: Source/Actions/Microsoft.Deployment.Actions.Salesforce/SalesforceSqlArtefacts.cs [hunk 6]
```diff
@@ -23,6 +23,8 @@ namespace Microsoft.Deployment.Actions.Salesforce
     [Export(typeof(IAction))]
     public class SalesforceSqlArtefacts : BaseAction
     {
+        List<string> coreObjects = new List<String>() { "Opportunity", "Account", "Lead", "Product2", "OpportunityLineItem", "OpportunityStage", "User", "UserRole" };
```
Comment (raymondb1) [inline Source/Actions/Microsoft.Deployment.Actions.Salesforce/SalesforceSqlArtefacts.cs]: Isn't this defined in the init.json

**Comment by cdoroftei** [comment]
addressed feedback requests

**Pull request merged**